lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -647,7 +648,16 @@ def _set_security_conf(self, data: PeerClusterRelData) -> None:
             self.charm.peers_data.put(Scope.APP, "security_index_initialised", True)
 
         if s3_creds := data.credentials.s3:
-            self.charm.secrets.put_object(Scope.APP, "s3-creds", s3_creds.to_dict(by_alias=True))
+            self.charm.secrets.put_object(
+                Scope.APP, S3_CREDENTIALS, s3_creds.to_dict(by_alias=True)
+            )
+        else:
+            # Set the S3 credentials to empty
+            self.charm.secrets.put_object(
+                Scope.APP,
+                S3_CREDENTIALS,
+                S3RelDataCredentials().to_dict(by_alias=True),
+            )
 
```

Comment:
can you explain why?

Reply:
This is the case we are missing s3 information, then we just create an empty object to signal that.